library/core/src/hint.rs

```diff
@@ -51,18 +51,52 @@ pub const unsafe fn unreachable_unchecked() -> ! {
     unsafe { intrinsics::unreachable() }
 }
 
-/// Emits a machine instruction hinting to the processor that it is running in busy-wait
-/// spin-loop ("spin lock").
+/// Emits a machine instruction to signal the processor that it is running in
+/// a busy-wait spin-loop ("spin lock").
+///
+/// Upon receiving the spin-loop signal the processor can optimize its behavior by,
+/// for example, saving power or switching hyper-threads.
+///
+/// This function is different from [`crate::thread::yield_now`] which directly
```

Comment:
I would not put `crate::` in the docs, I think `[thread::yield_now]` with a reference link `[thread::yield_now]: crate::thread::yield_now` would be easier to read, the way you did for `spin_loop_hint`.